dns-server/tests/basic_test.rs

```diff
@@ -85,6 +95,52 @@ pub async fn aaaa_crud() -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+#[tokio::test]
+pub async fn answers_match_question() -> Result<(), anyhow::Error> {
+    let test_ctx = init_client_server("not_wrong_record_answers").await?;
+    let client = &test_ctx.client;
+
+    // records should initially be empty
+    let records = dns_records_list(client, TEST_ZONE).await?;
+    assert!(records.is_empty());
+
+    // add an aaaa record
+    let name = "devron".to_string();
+    let addr = Ipv6Addr::new(0xfd, 0, 0, 0, 0, 0, 0, 0x1);
+    let aaaa = DnsRecord::Aaaa(addr);
+    let input_records = HashMap::from([(name.clone(), vec![aaaa])]);
+    dns_records_create(client, TEST_ZONE, input_records.clone()).await?;
+
+    // read back the aaaa record
+    let records = dns_records_list(client, TEST_ZONE).await?;
+    assert_eq!(input_records, records);
+
+    let name = Name::from_ascii(&(name + "." + TEST_ZONE + "."))
+        .expect("can construct name for query");
+
+    // it turns out `trust-dns-resolver`'s internal CachingClient will
+    // transparently correct the server misbehavior "server sent AAAA answers to
+    // an A query": the caching client will cache the extra record, then
+    // determine that there are no A records to respond with, and send a
+    // matching response.
+    //
+    // so, instead, `raw_dns_client_query` will let us see an actual server
+    // misbehavior, if one occurs.
+    let response = raw_dns_client_query(
+        test_ctx.dns_server.local_address(),
+        name,
+        RecordType::A,
+    )
+    .await
+    .expect("test query is ok");
+    assert_eq!(response.header().response_code(), ResponseCode::NoError);
+    assert_eq!(response.answers(), &[]);
+    assert_eq!(response.additionals(), &[]);
+
```

Comment:
What do you think about adding a case with an SRV lookup that produces AAAAs with additional records so that we verify that's still working?  edit: I guess we test this behavior already in srv_crud()?

Reply:
yeah, exactly, srv_crud() already covers that. i would kind of like to assert that the AAAAs are in Additionals _specifically_, but i didn't want to go too wild with `raw_dns_client_query` off the bat.